src/main/java/minijava/parser/Parser.java

```diff
@@ -310,16 +311,10 @@ private void parseExpressionWithPrecedenceClimbing(int minPrecedence) {
 
   /** UnaryExpression -> PostfixExpression | (! | -) UnaryExpression */
```

Comment:
You're correct. I thought that the (unecessary) recursion could become a problem in tests with lots of `!` in a row. But this isn't actually the case (newer test case shows this).